sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoBundle.java

```diff
@@ -150,11 +164,55 @@ public <T extends Repository<S, ID>, S, ID extends Serializable> T createReposit
     return factoryBean.getObject();
   }
 
+  /**
+   * creates a mongoTemplate to be used for operations in database. If the option {@code
+   * withValidation} was selected in the Bundle creation, it will initialize a spring application
+   * context for the validation
+   */
   private MongoOperations createMongoOperations() {
     SimpleMongoClientDatabaseFactory mongoDbFactory =
         new SimpleMongoClientDatabaseFactory(mongoClient, this.database);
     MongoConverter mongoConverter = getDefaultMongoConverter(mongoDbFactory, getConverters());
-    return new MongoTemplate(mongoDbFactory, mongoConverter);
+    var mongoTemplate = new MongoTemplate(mongoDbFactory, mongoConverter);
+    if (validationEnabled) {
+      mongoTemplate.setApplicationContext(applicationContext);
```

Comment:
Should we add a comment for future generation that we need the application context for the application events (`OnBeforeSave`) to be supported?